src/main/java/school/hei/haapi/endpoint/rest/mapper/StudentCourseMapper.java

```diff
@@ -0,0 +1,70 @@
+package school.hei.haapi.endpoint.rest.mapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import school.hei.haapi.endpoint.rest.model.StudentCourseExam;
+import school.hei.haapi.endpoint.rest.model.StudentExamGrade;
+import school.hei.haapi.model.Course;
+import school.hei.haapi.model.Grade;
+import school.hei.haapi.model.StudentCourse;
+import school.hei.haapi.model.User;
+import school.hei.haapi.service.CourseService;
+import school.hei.haapi.service.GradeService;
+import school.hei.haapi.service.UserService;
+
+@AllArgsConstructor
+@Component
+public class StudentCourseMapper {
+  private final UserService userService;
+  private final UserMapper userMapper;
+  private final GradeMapper gradeMapper;
+  private final GradeService gradeService;
```

Comment:
Calling Service inside Mapper is BAD ! 

Mapper must always map data.

If there is some grades that need to be mapped from service, just set it as arguments.

Do the same thing for all same case.